Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -654,12 +685,36 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
     } else {
         OWSAssert(indexPath.section == MessageComposeTableViewControllerSectionContacts)
 
-        NSString *identifier = [[[self contactForIndexPath:indexPath] textSecureIdentifiers] firstObject];
+            if (!self.searchController.active && self.hasNoContacts)
+        {
+            // TODO: We could do something here, like show the "invite contacts" UI.
+            return;
+        }
+
+        Contact *contact = [self contactForIndexPath:indexPath];
+        NSString *contactIdentifier = [[contact textSecureIdentifiers] firstObject];
+
+        __weak MessageComposeTableViewController *weakSelf = self;
+        void (^newContactConversationBlock)() = ^{
+            [weakSelf dismissViewControllerAnimated:YES
+                                         completion:^() {
+                                             [Environment messageIdentifier:contactIdentifier withCompose:YES];
+                                         }];
+        };
 
-        [self dismissViewControllerAnimated:YES
-                                 completion:^() {
-                                     [Environment messageIdentifier:identifier withCompose:YES];
-                                 }];
+        if ([self isContactBlocked:contact]) {
```

Comment:
Point taken, I've reverted this change.  But I'm going to leave similar logic in the "new/edit group" view, since a) group members are unblocked by sending messages, etc. b) the "search" argument isn't relevant either.